x-pack/plugins/observability/public/components/shared/exploratory_view/hooks/use_app_index_pattern.tsx

```diff
@@ -41,33 +40,41 @@ export function IndexPatternContextProvider({ children }: ProviderProps) {
     synthetics: null,
     ux: null,
     apm: null,
+    mobile: null,
   } as HasAppDataState);
 
   const {
     services: { data },
   } = useKibana<ObservabilityPublicPluginsStart>();
 
-  const checkIfAppHasData = async (dataType: AppDataType) => {
-    const handler = getDataHandler(dataType);
-    return handler?.hasData();
-  };
-
   const loadIndexPattern: IIndexPatternContext['loadIndexPattern'] = useCallback(
     async ({ dataType }) => {
       setSelectedApp(dataType);
 
       if (hasAppData[dataType] === null) {
         setLoading(true);
         try {
-          const hasDataResponse = (await checkIfAppHasData(dataType)) as HasDataResponse;
-
-          const hasDataT = hasDataResponse.hasData;
-
+          let hasDataT = false;
+          let indices: string | undefined = '';
+          switch (dataType) {
+            case 'ux':
+            case 'synthetics':
+              const resultUx = await getDataHandler(dataType)?.hasData();
+              hasDataT = Boolean(resultUx?.hasData);
+              indices = resultUx?.indices;
+              break;
+            case 'apm':
+            case 'mobile':
+              const resultApm = await getDataHandler('apm')?.hasData();
+              hasDataT = Boolean(resultApm?.hasData);
+              indices = resultApm?.indices['apm_oss.transactionIndices'];
```

Comment:
In 7.14 we are starting to move towards a "metrics powered ui". This means that APM Server might start dropping transactions, and they can therefore only be used for samples - not representative of overall averages or anything like that. 

Instead you'll have to query the metrics index. 

For 7.14 this is optional so in APM we are conditionally querying metrics if they exists, else falling back to transactions.

Reply:
This might not be something you have to worry about too much until 7.15 but just an fyi that this will break eventually

Reply:
That's a very good point @sqren ! We should actually discuss whether dropping transactions for mobile apps is meaningful at all, since analytics use cases (slicing the data by many different dimensions like Geo-location, device type, device versions, service versions, etc.) are very important in that area, which won't be easily doable based on metrics. Or maybe we need to generate separate data out of it (something similar to the RUM data).